capstone-backend/src/main/java/com/google/sps/servlets/Utility.java

```diff
@@ -349,4 +352,116 @@ else if (key.equals("remove_"+name)) {
+    // Every requiredField must be present as a key in the jsonObject. If not, this check will fail. 
+    if (list.size() != requiredFields.size()) {
+      System.err.println("Error: Not all required fields are present in the request body, or some required fields are empty.");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return false;
+    } 
+    return true;
+  }
+
+  /**
+   * This method executes the given post request, creating the new object and returning it with its ID appended.
+   * @param {collectionReference} reference to the appropriate database collection
+   * @param {request} the request sent to the backend
+   * @param {genericClass} a Generic class, to be used in casting objects retrieved from the database
+   * @param {requiredFields} a list of the names of fields deemed "necessary" for creation of an object.
+   * @return T the object created by the post request, with an ID field generated by firebase
+   */
+  public static <T extends BaseEntity> T post(CollectionReference collectionReference, HttpServletRequest request, 
+      HttpServletResponse response, GenericClass<T> genericClass, List<String> requiredFields) throws IOException {
+    Map<String, Object> constructorFields = new HashMap<>();
+    JsonObject jsonObject = Utility.createRequestBodyJson(request);
+    Field[] fields = genericClass.getMyType().getDeclaredFields();
+    if (!postErrorHandler(jsonObject, response, fields, requiredFields)) {
+      return null;
+    }
+    for (Field f : fields) {
+      String fName = f.getName();
+      String type = f.getType().getName();
+      if (type.equals("int") || type.equals("java.lang.Integer")) {
+        if (jsonObject.has(fName) && jsonObject.get(fName).getAsString().length() != 0) {
+          constructorFields.put(name, jsonObject.get(fName).getAsInt());
+        }
+        else {
+          constructorFields.put(fName, -1);
+        }
+      }
+      else if (type.equals("java.lang.String")) {
+        if (jsonObject.has(fName) && jsonObject.get(fName).getAsString().length() != 0) {
+          constructorFields.put(fName, jsonObject.get(fName).getAsString());
+        }
+        else {
+          constructorFields.put(fName, "");
+        }
+      }
+      else if (type.equals("boolean") || type.equals("java.lang.boolean")) {
+        if (jsonObject.has(fName) && jsonObject.get(fName).getAsString().length() != 0) {
+          constructorFields.put(fName, jsonObject.get(fName).getAsBoolean());
+        }
+        else {
+          constructorFields.put(fName, false);
+        }
+      }
+      else if (type.equals("java.util.List") || type.equals("java.util.ArrayList")) {
+        if (jsonObject.has(fName) && jsonObject.get(fName).getAsString().length() != 0) {
+          // First check to see if it's an "actual array"
+          try {
+            JsonElement value = jsonObject.get(fName);
+            Type listType = new TypeToken<List<String>>() {}.getType();
```

Comment:
Nice job figuring out how to parse a JsonElement into a list!

Reply:
Thanks!